### PR TITLE
Added the Check RTT function to check if the device is live on network

### PR DESCRIPTION
**Pull request by nitu-s-gupta**
Description
RTT is used to continously ping to devices discovered by HE to get pong response. If HE fails to receive response from any particular device, then it waits for 12 retries and then finally deletes the device from the db.

Steps to check:
1. Launch the android HE
2. Start the Linux HE
3. Linux HE shall be discovered by Android HE

```
D/Utils: id : edge-orchestration-53b41e4c-bd9f-422a-aa4a-1adb71ef3c65
D/NsdHelper: The DeviceID generated is edge-orchestration-53b41e4c-bd9f-422a-aa4a-1adb71ef3c65
D/NsdHelper: Initialize NSD
D/NsdHelper: Calling Service Discovery
D/NsdHelper: Service discovery started
D/HomeEdgeService: Wifi states has changed
D/Utils: The IP resolved is 192.168.1.109
D/HomeEdgeService: Starting server
D/HTTPServer: Routes are added
D/HTTPServer: Setting the hostname as 192.168.1.109port to : 56001
D/HomeEdgeService: Staring the Server
D/HomeEdgeService: The Server is running on 192.168.1.109
D/HTTPServer: Routes are added
D/HTTPServer: Setting the hostname as 192.168.1.109port to : 56002
D/HomeEdgeService: Staring the Server
I/TcpOptimizer: [insertTCPSessionList] sid=101, g_tcpConnListCount=1
I/TcpOptimizer: [listen] sid=101, port=56002
I/TcpOptimizer: [deleteTCPSessionList] sid=101, g_tcpConnListCount=0
D/HomeEdgeService: The Server is running on 192.168.1.109
D/HomeEdgeService: Wifi states has changed
D/Utils: The IP resolved is 192.168.1.109
D/HomeEdgeService: HTTP Server is not null
D/HomeEdgeService: The IP now is 192.168.1.109
D/RTT: Device List size is 0
W/System: A resource failed to call close. 
W/System: A resource failed to call close. 
I/TcpOptimizer: [tcpStateMonitor] next... nSockets=1
I/TcpOptimizer: [tcpStateMonitor] Pause
D/RTT: Device List size is 0
D/RTT: Device List size is 0
D/RTT: Device List size is 0
D/RTT: Device List size is 0
D/RTT: Device List size is 0
D/RTT: Device List size is 0
D/RTT: Device List size is 0
D/RTT: Device List size is 0
D/NsdHelper: Service discovery successname: edge-orchestration-314b6cd6-a8af-46ea-a3a8-452495c8b02e, type: _orchestration._tcp., host: null, port: 0, txtRecord: 
D/NsdHelper: Resolving the service
D/NsdHelper: Resolve Succeeded. name: edge-orchestration-314b6cd6-a8af-46ea-a3a8-452495c8b02e, type: ._orchestration._tcp, host: /192.168.1.110, port: 42425, txtRecord:  AISER= Platform=docker ExecType=container
D/NsdHelper: Exectype:container
D/NsdHelper: Platform:docker
D/NsdHelper: Creating a device to insert in DB
I/lfedge.homeedg: Waiting for a blocking GC ClassLinker
D/NsdHelper: Calling Orchestration API
D/APIImpl: getOrchestrationInfo is getting called at http://192.168.1.110:56002/api/v1/discoverymgr/orchestrationinfo
```
4. Stop the Linux HE to see if it gets removed from the Android HE Db

```
D/RTT: Device List size is 1
D/RTT: Device obtained isedge-orchestration-314b6cd6-a8af-46ea-a3a8-452495c8b02e
I/TcpOptimizer: Restarting tcpStateMonitor
I/TcpOptimizer: [insertTCPSessionList] sid=6, g_tcpConnListCount=1
I/TcpOptimizer: [tcpStateMonitor] Resume
I/TcpOptimizer: [updateIPandPort] (192.168.1.110_56002), sid=6
D/APIImpl: Pong

D/RTT: Device List size is 1
D/RTT: Device obtained isedge-orchestration-314b6cd6-a8af-46ea-a3a8-452495c8b02e
W/System.err: java.net.ConnectException: Failed to connect to /192.168.1.110:56002
W/System.err:     at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:265)
W/System.err:     at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:183)
W/System.err:     at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.java:224)
W/System.err:     at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.java:108)
W/System.err:     at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.java:88)
W/System.err:     at okhttp3.internal.connection.Transmitter.newExchange(Transmitter.java:169)
W/System.err:     at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:41)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
W/System.err:     at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:94)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
W/System.err:     at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
W/System.err:     at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:88)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
W/System.err:     at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:229)
W/System.err:     at okhttp3.RealCall.execute(RealCall.java:81)
W/System.err:     at retrofit2.OkHttpCall.execute(OkHttpCall.java:204)
W/System.err:     at retrofit2.DefaultCallAdapterFactory$ExecutorCallbackCall.execute(DefaultCallAdapterFactory.java:108)
W/System.err:     at org.lfedge.homeedge.client.APIImpl.getPing(APIImpl.java:137)
W/System.err:     at org.lfedge.homeedge.common.RoundTripTime.run(RoundTripTime.java:37)
W/System.err:     at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
W/System.err:     at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:307)
W/System.err:     at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:302)
W/System.err:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
W/System.err:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
W/System.err:     at java.lang.Thread.run(Thread.java:923)
W/System.err: Caused by: java.net.ConnectException: failed to connect to /192.168.1.110 (port 56002) from /192.168.1.109 (port 60612) after 10000ms: isConnected failed: ECONNREFUSED (Connection refused)
W/System.err:     at libcore.io.IoBridge.isConnected(IoBridge.java:287)
W/System.err:     at libcore.io.IoBridge.connectErrno(IoBridge.java:192)
W/System.err:     at libcore.io.IoBridge.connect(IoBridge.java:134)
W/System.err:     at java.net.PlainSocketImpl.socketConnect(PlainSocketImpl.java:142)
W/System.err:     at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:390)
W/System.err:     at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:230)
W/System.err:     at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:212)
W/System.err:     at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:436)
W/System.err:     at java.net.Socket.connect(Socket.java:621)
W/System.err:     at okhttp3.internal.platform.AndroidPlatform.connectSocket(AndroidPlatform.java:71)
W/System.err:     at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:263)
W/System.err: 	... 28 more
W/System.err: Caused by: android.system.ErrnoException: isConnected failed: ECONNREFUSED (Connection refused)
W/System.err:     at libcore.io.IoBridge.isConnected(IoBridge.java:274)
W/System.err: 	... 38 more

D/RTT: Deleting the deviceedge-orchestration-314b6cd6-a8af-46ea-a3a8-452495c8b02e
D/RTT: Device List size is 0
```

### CODE REVIEW

File: app/src/main/java/org/lfedge/homeedge/common/RoundTripTime.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            Log.d(TAG,"Device obtained is" +device.getEdge_orch_id());
+            Utils.setOrchestrationInfoAPI(device.getIp());
+            String targetURL = Utils.BASE_URL + Utils.API.PING_REQUEST;
+            APIImpl api = new APIImpl(mContext);
+            String response = null;
+            try {
+                int tryLimit = 12;
+                while(response==null && tryLimit>0){
+                    response = api.getPing(targetURL);
+                    tryLimit--;
```
Comment (smarthome-hanium) [inline app/src/main/java/org/lfedge/homeedge/common/RoundTripTime.java:38]: What about sleeping 1 sec if getPing() returns null?

Comment (tiokim) [inline app/src/main/java/org/lfedge/homeedge/common/RoundTripTime.java:38]: What about sleeping 1 sec if getPing() returns null?

Comment (nitu-s-gupta) [reply app/src/main/java/org/lfedge/homeedge/common/RoundTripTime.java:38]: It is in sync with Linux HE and it tries for 12 times before deleting the device

Comment (tiokim) [reply app/src/main/java/org/lfedge/homeedge/common/RoundTripTime.java:38]: It tried 12 times in a row without waiting. 
As I know, Linux HE sleeps `defaultRttDuration` time. Please see the [code](https://github.com/lf-edge/edge-home-orchestration-go/blob/d96c2797bf23933e00a9788320052e45b2689054/internal/common/resourceutil/roundtriptime.go#L81).

**Review by tiokim — APPROVED** [review]

**Pull request merged**